crates/cli/src/plugins.rs

```diff
@@ -1,86 +1,113 @@
 use crate::bytecode;
 use anyhow::{anyhow, bail, Result};
-use serde::Deserialize;
 use std::{
     fs,
-    io::{Read, Seek},
+    io::{self},
     path::Path,
     str,
 };
 use walrus::{ExportItem, ValType};
-use wasi_common::{pipe::WritePipe, sync::WasiCtxBuilder};
-use wasmtime::{AsContextMut, Engine, Linker};
 use wizer::Wizer;
 
-const PLUGIN_MODULE: &[u8] = include_bytes!(concat!(env!("OUT_DIR"), "/plugin.wasm"));
+/// Represents the kind of plugin.
+#[cfg(feature = "plugin-internal")]
+#[derive(Clone, Copy, Debug)]
+pub(crate) enum PluginKind {
+    // Built-in system plugins (e.g., default, legacy).
+    Internal(InternalPluginKind),
+    // External plugins are provided by a user.
+    External,
+}
 
-/// Use the legacy plugin when using the `compile -d` command.
-const QUICKJS_PROVIDER_V2_MODULE: &[u8] = include_bytes!("./javy_quickjs_provider_v2.wasm");
+/// Represents the kind of plugin.
+#[cfg(not(feature = "plugin-internal"))]
+#[derive(Clone, Copy, Debug)]
+enum PluginKind {
+    // Built-in system plugins (e.g., default, legacy).
+    Internal(InternalPlugin),
+    // External plugins are provided by a user.
+    External,
+}
 
-/// A property that is in the config schema returned by the plugin.
-#[derive(Debug, Deserialize)]
-#[serde(rename_all = "camelCase")]
-pub(crate) struct JsConfigProperty {
-    /// The name of the property (e.g., `simd-json-builtins`).
-    pub(crate) name: String,
-    /// The documentation to display for the property.
-    pub(crate) doc: String,
+/// Represents the kind of internal plugin.
+#[cfg(feature = "plugin-internal")]
+#[derive(Clone, Copy, Debug)]
+pub(crate) enum InternalPluginKind {
+    // Default internal plugin.
+    Default,
+    // Legacy V2 internal plugin.
+    V2,
 }
 
-/// Different plugins that are available.
-#[derive(Debug)]
-pub enum Plugin {
-    /// The default plugin.
+/// Represents the kind of internal plugin.
+#[cfg(not(feature = "plugin-internal"))]
+#[derive(Clone, Copy, Debug)]
+enum InternalPluginKind {
+    // Default internal plugin.
     Default,
-    /// A plugin for use with the `compile` to maintain backward compatibility.
+    // Legacy V2 internal plugin.
     V2,
-    /// A plugin provided by the user.
-    User { bytes: Vec<u8> },
 }
 
-impl Default for Plugin {
-    fn default() -> Self {
-        Self::Default
-    }
+/// Represents any valid Javy plugin.
+#[derive(Clone, Debug)]
+pub(crate) struct Plugin {
+    pub(crate) bytes: Vec<u8>,
+    kind: PluginKind,
```

Comment:
Instead of this, can we do:

```suggestion
/// Javy plugin.
pub struct Plugin(Vec<u8>);

// Represents any valid Javy plugin.
#[derive(Clone, Debug)]
pub(crate) struct CliPlugin {
    pub(crate) bytes: Plugin,
    kind: PluginKind,
```

With the idea that the `Plugin` struct will live in the codegen crate and the `CliPlugin` will live in the CLI crate the `Generator` will reference `Plugin`. Basically I want to keep the `PluginKind` knowledge entirely in the CLI crate and use a couple feature gated booleans to get `Generator` to do what it needs to do rather than use a richer plugin abstraction in the generator so it's straightforward for a codegen user to just pass their plugin bytes along without having to worry about `PluginKind`s.